Failing firedrake-parmmg docker build (HDF5 and ParMETIS)

The build failed today: https://github.com/mesh-adaptation/docs/actions/runs/14676496288/job/41193457713

This is the issue it reports, related to HDF5:
```
#10 24.09   Unable to find hdf5 in default locations!
#10 24.09   Perhaps you can specify with --with-hdf5-dir=<directory>
#10 24.09   If you do not want hdf5, then give --with-hdf5=0
#10 24.09   You might also consider using --download-hdf5 instead
```
Then after I pass `--download-hdf5` it raises another error for ParMETIS (in `petsc/configure.log`):
```
In file included from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers/GKlib.h:66,
                 from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/./parmetislib.h:19,
                 from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/akwayfm.c:14:
/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers/gk_externs.h:20:10: fatal error: gklib_tls.h: No such file or directory
   20 | #include <gklib_tls.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
```
<details><summary>Full traceback</summary>
<p>
=============================================================================================
               Compiling and installing PARMETIS; this may take several minutes
=============================================================================================
Executing: /usr/bin/make -j4 -l18.0
stdout:
/usr/bin/cmake -S/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis -B/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build/CMakeFiles /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build//CMakeFiles/progress.marks
/usr/bin/make  -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build'
/usr/bin/make  -f libparmetis/CMakeFiles/parmetis.dir/build.make libparmetis/CMakeFiles/parmetis.dir/depend
make[2]: Entering directory '/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build'
cd /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build/libparmetis /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build/libparmetis/CMakeFiles/parmetis.dir/DependInfo.cmake "--color="
make[2]: Leaving directory '/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build'
/usr/bin/make  -f libparmetis/CMakeFiles/parmetis.dir/build.make libparmetis/CMakeFiles/parmetis.dir/build
make[2]: Entering directory '/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build'
[  4%] Building C object libparmetis/CMakeFiles/parmetis.dir/ametis.c.o
[  4%] Building C object libparmetis/CMakeFiles/parmetis.dir/akwayfm.c.o
cd /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build/libparmetis && /usr/bin/mpicc -Dparmetis_EXPORTS -I/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/include -I/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers -I/include -I/home/firedrake/petsc/arch-firedrake-default/include -I/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/. -fPIC -Wno-lto-type-mismatch -Wno-stringop-overflow -O3 -mtune=generic -I/usr/include/hdf5/openmpi -I/usr/include/scotch -I/usr/include/superlu -I/usr/include/superlu-dist -L/usr/lib/x86_64-linux-gnu/hdf5/openmpi -fPIC -Wno-lto-type-mismatch -Wno-stringop-overflow -O3 -mtune=generic -I/usr/include/hdf5/openmpi -I/usr/include/scotch -I/usr/include/superlu -I/usr/include/superlu-dist -L/usr/lib/x86_64-linux-gnu/hdf5/openmpi -fPIC -MD -MT libparmetis/CMakeFiles/parmetis.dir/akwayfm.c.o -MF CMakeFiles/parmetis.dir/akwayfm.c.o.d -o CMakeFiles/parmetis.dir/akwayfm.c.o -c /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/akwayfm.c
[  7%] Building C object libparmetis/CMakeFiles/parmetis.dir/ComputeVertexSeparator.c.o
cd /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build/libparmetis && /usr/bin/mpicc -Dparmetis_EXPORTS -I/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/include -I/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers -I/include -I/home/firedrake/petsc/arch-firedrake-default/include -I/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/. -fPIC -Wno-lto-type-mismatch -Wno-stringop-overflow -O3 -mtune=generic -I/usr/include/hdf5/openmpi -I/usr/include/scotch -I/usr/include/superlu -I/usr/include/superlu-dist -L/usr/lib/x86_64-linux-gnu/hdf5/openmpi -fPIC -Wno-lto-type-mismatch -Wno-stringop-overflow -O3 -mtune=generic -I/usr/include/hdf5/openmpi -I/usr/include/scotch -I/usr/include/superlu -I/usr/include/superlu-dist -L/usr/lib/x86_64-linux-gnu/hdf5/openmpi -fPIC -MD -MT libparmetis/CMakeFiles/parmetis.dir/ametis.c.o -MF CMakeFiles/parmetis.dir/ametis.c.o.d -o CMakeFiles/parmetis.dir/ametis.c.o -c /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/ametis.c
cd /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build/libparmetis && /usr/bin/mpicc -Dparmetis_EXPORTS -I/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/include -I/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers -I/include -I/home/firedrake/petsc/arch-firedrake-default/include -I/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/. -fPIC -Wno-lto-type-mismatch -Wno-stringop-overflow -O3 -mtune=generic -I/usr/include/hdf5/openmpi -I/usr/include/scotch -I/usr/include/superlu -I/usr/include/superlu-dist -L/usr/lib/x86_64-linux-gnu/hdf5/openmpi -fPIC -Wno-lto-type-mismatch -Wno-stringop-overflow -O3 -mtune=generic -I/usr/include/hdf5/openmpi -I/usr/include/scotch -I/usr/include/superlu -I/usr/include/superlu-dist -L/usr/lib/x86_64-linux-gnu/hdf5/openmpi -fPIC -MD -MT libparmetis/CMakeFiles/parmetis.dir/ComputeVertexSeparator.c.o -MF CMakeFiles/parmetis.dir/ComputeVertexSeparator.c.o.d -o CMakeFiles/parmetis.dir/ComputeVertexSeparator.c.o -c /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/ComputeVertexSeparator.c
[  9%] Building C object libparmetis/CMakeFiles/parmetis.dir/balancemylink.c.o
cd /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build/libparmetis && /usr/bin/mpicc -Dparmetis_EXPORTS -I/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/include -I/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers -I/include -I/home/firedrake/petsc/arch-firedrake-default/include -I/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/. -fPIC -Wno-lto-type-mismatch -Wno-stringop-overflow -O3 -mtune=generic -I/usr/include/hdf5/openmpi -I/usr/include/scotch -I/usr/include/superlu -I/usr/include/superlu-dist -L/usr/lib/x86_64-linux-gnu/hdf5/openmpi -fPIC -Wno-lto-type-mismatch -Wno-stringop-overflow -O3 -mtune=generic -I/usr/include/hdf5/openmpi -I/usr/include/scotch -I/usr/include/superlu -I/usr/include/superlu-dist -L/usr/lib/x86_64-linux-gnu/hdf5/openmpi -fPIC -MD -MT libparmetis/CMakeFiles/parmetis.dir/balancemylink.c.o -MF CMakeFiles/parmetis.dir/balancemylink.c.o.d -o CMakeFiles/parmetis.dir/balancemylink.c.o -c /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/balancemylink.c
make[2]: Leaving directory '/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build'
make[1]: Leaving directory '/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build'
stdout:
In file included from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers/GKlib.h:66,
                 from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/./parmetislib.h:19,
                 from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/akwayfm.c:14:
/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers/gk_externs.h:20:10: fatal error: gklib_tls.h: No such file or directory
   20 | #include <gklib_tls.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
In file included from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers/GKlib.h:66,
                 from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/./parmetislib.h:19,
                 from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/balancemylink.c:14:
/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers/gk_externs.h:20:10: fatal error: gklib_tls.h: No such file or directory
   20 | #include <gklib_tls.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[2]: *** [libparmetis/CMakeFiles/parmetis.dir/build.make:93: libparmetis/CMakeFiles/parmetis.dir/akwayfm.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [libparmetis/CMakeFiles/parmetis.dir/build.make:121: libparmetis/CMakeFiles/parmetis.dir/balancemylink.c.o] Error 1
In file included from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers/GKlib.h:66,
                 from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/./parmetislib.h:19,
                 from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/ComputeVertexSeparator.c:5:
/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers/gk_externs.h:20:10: fatal error: gklib_tls.h: No such file or directory
   20 | #include <gklib_tls.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
In file included from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers/GKlib.h:66,
                 from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/./parmetislib.h:19,
                 from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/ametis.c:15:
/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers/gk_externs.h:20:10: fatal error: gklib_tls.h: No such file or directory
   20 | #include <gklib_tls.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[2]: *** [libparmetis/CMakeFiles/parmetis.dir/build.make:79: libparmetis/CMakeFiles/parmetis.dir/ComputeVertexSeparator.c.o] Error 1
make[2]: *** [libparmetis/CMakeFiles/parmetis.dir/build.make:107: libparmetis/CMakeFiles/parmetis.dir/ametis.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:119: libparmetis/CMakeFiles/parmetis.dir/all] Error 2
make: *** [Makefile:139: all] Error 2
                    Error running make on  PARMETIS: Could not execute "['/usr/bin/make -j4 -l18.0  ']":
/usr/bin/cmake -S/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis -B/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_pr
 .... more output .....
parmetis/balancemylink.c
make[2]: Leaving directory '/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build'
make[1]: Leaving directory '/home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/petsc-build'In file included from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/headers/GKlib.h:66,
                 from /home/firedrake/petsc/arch-firedrake-default/externalpackages/git.parmetis/libparmetis/./parmetislib.h:19,

 .... more error .....
rror 1
make[2]: *** [libparmetis/CMakeFiles/parmetis.dir/build.make:107: libparmetis/CMakeFiles/parmetis.dir/ametis.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:119: libparmetis/CMakeFiles/parmetis.dir/all] Error 2
make: *** [Makefile:139: all] Error 2

**** Configure header /tmp/petsc-jj4e0cvh/confdefs.h ****
#if !defined(INCLUDED_UNKNOWN)
#define INCLUDED_UNKNOWN

#define PETSC_CLANGUAGE_C 1
#define PETSC_CXX_RESTRICT __restrict
#define PETSC_FORTRAN_CHARLEN_T size_t
#define PETSC_FORTRAN_TYPE_INITIALIZE  = -2
#define PETSC_HAVE_ACCESS 1
#define PETSC_HAVE_BZERO 1
#define PETSC_HAVE_C99_COMPLEX 1
#define PETSC_HAVE_CLOCK 1
#define PETSC_HAVE_CXX_ATOMIC 1
#define PETSC_HAVE_CXX_COMPLEX 1
#define PETSC_HAVE_CXX_COMPLEX_FIX 1
#define PETSC_HAVE_CXX_DIALECT_CXX11 1
#define PETSC_HAVE_CXX_DIALECT_CXX14 1
#define PETSC_HAVE_CXX_DIALECT_CXX17 1
#define PETSC_HAVE_CXX_DIALECT_CXX20 1
#define PETSC_HAVE_DLADDR 1
#define PETSC_HAVE_DLCLOSE 1
#define PETSC_HAVE_DLERROR 1
#define PETSC_HAVE_DLFCN_H 1
#define PETSC_HAVE_DLOPEN 1
#define PETSC_HAVE_DLSYM 1
#define PETSC_HAVE_DRAND48 1
#define PETSC_HAVE_DYNAMIC_LIBRARIES 1
#define PETSC_HAVE_ERF 1
#define PETSC_HAVE_EXECUTABLE_EXPORT 1
#define PETSC_HAVE_FCNTL_H 1
#define PETSC_HAVE_FENV_H 1
#define PETSC_HAVE_FE_VALUES 1
#define PETSC_HAVE_FLOAT_H 1
#define PETSC_HAVE_FORK 1
#define PETSC_HAVE_FORTRAN_FLUSH 1
#define PETSC_HAVE_FORTRAN_FREE_LINE_LENGTH_NONE 1
#define PETSC_HAVE_FORTRAN_TYPE_STAR 1
#define PETSC_HAVE_FORTRAN_UNDERSCORE 1
#define PETSC_HAVE_GETCWD 1
#define PETSC_HAVE_GETDOMAINNAME 1
#define PETSC_HAVE_GETHOSTBYNAME 1
#define PETSC_HAVE_GETHOSTNAME 1
#define PETSC_HAVE_GETPAGESIZE 1
#define PETSC_HAVE_GETRUSAGE 1
#define PETSC_HAVE_IMMINTRIN_H 1
#define PETSC_HAVE_INTTYPES_H 1
#define PETSC_HAVE_ISINF 1
#define PETSC_HAVE_ISNAN 1
#define PETSC_HAVE_ISNORMAL 1
#define PETSC_HAVE_LGAMMA 1
#define PETSC_HAVE_LINUX 1
#define PETSC_HAVE_LOG2 1
#define PETSC_HAVE_LSEEK 1
#define PETSC_HAVE_MALLOC_H 1
#define PETSC_HAVE_MEMMOVE 1
#define PETSC_HAVE_MKSTEMP 1
#define PETSC_HAVE_MPIEXEC_ENVIRONMENTAL_VARIABLE OMP
#define PETSC_HAVE_MPIIO 1
#define PETSC_HAVE_MPI_COMBINER_CONTIGUOUS 1
#define PETSC_HAVE_MPI_COMBINER_DUP 1
#define PETSC_HAVE_MPI_COMBINER_NAMED 1
#define PETSC_HAVE_MPI_COUNT 1
#define PETSC_HAVE_MPI_F90MODULE 1
#define PETSC_HAVE_MPI_F90MODULE_VISIBILITY 1
#define PETSC_HAVE_MPI_GET_ACCUMULATE 1
#define PETSC_HAVE_MPI_GET_LIBRARY_VERSION 1
#define PETSC_HAVE_MPI_INIT_THREAD 1
#define PETSC_HAVE_MPI_INT64_T 1
#define PETSC_HAVE_MPI_LONG_DOUBLE 1
#define PETSC_HAVE_MPI_NEIGHBORHOOD_COLLECTIVES 1
#define PETSC_HAVE_MPI_NONBLOCKING_COLLECTIVES 1
#define PETSC_HAVE_MPI_ONE_SIDED 1
#define PETSC_HAVE_MPI_PROCESS_SHARED_MEMORY 1
#define PETSC_HAVE_MPI_REDUCE_LOCAL 1
#define PETSC_HAVE_MPI_REDUCE_SCATTER_BLOCK 1
#define PETSC_HAVE_MPI_RGET 1
#define PETSC_HAVE_MPI_WIN_CREATE 1
#define PETSC_HAVE_NANOSLEEP 1
#define PETSC_HAVE_NETDB_H 1
#define PETSC_HAVE_NETINET_IN_H 1
#define PETSC_HAVE_NO_FINITE_MATH_ONLY 1
#define PETSC_HAVE_OPENMPI 1
#define PETSC_HAVE_POPEN 1
#define PETSC_HAVE_POSIX_MEMALIGN 1
#define PETSC_HAVE_PTHREAD_MUTEX 1
#define PETSC_HAVE_PWD_H 1
#define PETSC_HAVE_RAND 1
#define PETSC_HAVE_READLINK 1
#define PETSC_HAVE_REALPATH 1
#define PETSC_HAVE_REAL___FLOAT128 1
#define PETSC_HAVE_RTLD_GLOBAL 1
#define PETSC_HAVE_RTLD_LAZY 1
#define PETSC_HAVE_RTLD_LOCAL 1
#define PETSC_HAVE_RTLD_NOW 1
#define PETSC_HAVE_SETJMP_H 1
#define PETSC_HAVE_SHMGET 1
#define PETSC_HAVE_SLEEP 1
#define PETSC_HAVE_SNPRINTF 1
#define PETSC_HAVE_SOCKET 1
#define PETSC_HAVE_STDATOMIC_H 1
#define PETSC_HAVE_STDINT_H 1
#define PETSC_HAVE_STRCASECMP 1
#define PETSC_HAVE_STRINGS_H 1
#define PETSC_HAVE_STRUCT_SIGACTION 1
#define PETSC_HAVE_SYS_PARAM_H 1
#define PETSC_HAVE_SYS_PROCFS_H 1
#define PETSC_HAVE_SYS_RESOURCE_H 1
#define PETSC_HAVE_SYS_SOCKET_H 1
#define PETSC_HAVE_SYS_TIMES_H 1
#define PETSC_HAVE_SYS_TIME_H 1
#define PETSC_HAVE_SYS_TYPES_H 1
#define PETSC_HAVE_SYS_UTSNAME_H 1
#define PETSC_HAVE_SYS_WAIT_H 1
#define PETSC_HAVE_TGAMMA 1
#define PETSC_HAVE_TIME 1
#define PETSC_HAVE_TIME_H 1
#define PETSC_HAVE_UNAME 1
#define PETSC_HAVE_UNISTD_H 1
#define PETSC_HAVE_USLEEP 1
#define PETSC_HAVE_VA_COPY 1
#define PETSC_HAVE_VSNPRINTF 1
#define PETSC_IS_COLORING_MAX USHRT_MAX
#define PETSC_IS_COLORING_VALUE_TYPE short
#define PETSC_IS_COLORING_VALUE_TYPE_F integer2
#define PETSC_LEVEL1_DCACHE_LINESIZE 64
#define PETSC_MAX_PATH_LEN 4096
#define PETSC_MEMALIGN 16
#define PETSC_MPIU_IS_COLORING_VALUE_TYPE MPI_UNSIGNED_SHORT
#define PETSC_OMAKE "/usr/bin/make --no-print-directory"
#define PETSC_PYTHON_EXE "/usr/bin/python3"
#define PETSC_SIGNAL_CAST
#define PETSC_SIZEOF_INT 4
#define PETSC_SIZEOF_LONG 8
#define PETSC_SIZEOF_LONG_LONG 8
#define PETSC_SIZEOF_SIZE_T 8
#define PETSC_SIZEOF_VOID_P 8
#define PETSC_USE_AVX512_KERNELS 1
#define PETSC_USE_BACKWARD_LOOP 1
#define PETSC_USE_CTABLE 1
#define PETSC_USE_DMLANDAU_2D 1
#define PETSC_USE_INFO 1
#define PETSC_USE_LOG 1
#define PETSC_USE_MALLOC_COALESCED 1
#define PETSC_USE_PROC_FOR_SIZE 1
#define PETSC_USE_REAL_DOUBLE 1
#define PETSC_USE_SHARED_LIBRARIES 1
#define PETSC_USE_VISIBILITY_C 1
#define PETSC_USE_VISIBILITY_CXX 1
#define PETSC_USING_64BIT_PTR 1
#define PETSC_USING_F2003 1
#define PETSC_USING_F90FREEFORM 1
#define PETSC_VERSION_BRANCH_GIT "HEAD"
#define PETSC_VERSION_DATE_GIT "2025-03-29 00:32:40 -0500"
#define PETSC_VERSION_GIT "v3.23.0"
#define PETSC__BSD_SOURCE 1
#define PETSC__DEFAULT_SOURCE 1
#define PETSC__GNU_SOURCE 1
#endif
**** C specific Configure header /tmp/petsc-jj4e0cvh/conffix.h ****
#if !defined(INCLUDED_UNKNOWN)
#define INCLUDED_UNKNOWN

#if defined(__cplusplus)
extern "C" {
}
#else
#endif
#endif
*********************************************************************************************
           UNABLE to CONFIGURE with GIVEN OPTIONS (see configure.log for details):
---------------------------------------------------------------------------------------------
                               Error running make on  PARMETIS
*********************************************************************************************
  File "/home/firedrake/petsc/config/configure.py", line 461, in petsc_configure
    framework.configure(out = sys.stdout)
  File "/home/firedrake/petsc/config/BuildSystem/config/framework.py", line 1462, in configure
    self.processChildren()
  File "/home/firedrake/petsc/config/BuildSystem/config/framework.py", line 1450, in processChildren
    self.serialEvaluation(self.childGraph)
  File "/home/firedrake/petsc/config/BuildSystem/config/framework.py", line 1425, in serialEvaluation
    child.configure()
  File "/home/firedrake/petsc/config/BuildSystem/config/package.py", line 1354, in configure
    self.executeTest(self.configureLibrary)
  File "/home/firedrake/petsc/config/BuildSystem/config/base.py", line 138, in executeTest
    ret = test(*args,**kargs)
          ^^^^^^^^^^^^^^^^^^^
  File "/home/firedrake/petsc/config/BuildSystem/config/packages/parmetis.py", line 43, in configureLibrary
    config.package.Package.configureLibrary(self)
  File "/home/firedrake/petsc/config/BuildSystem/config/package.py", line 1041, in configureLibrary
    for location, directory, lib, incl in self.generateGuesses():
  File "/home/firedrake/petsc/config/BuildSystem/config/package.py", line 609, in generateGuesses
    d = self.checkDownload()
        ^^^^^^^^^^^^^^^^^^^^
  File "/home/firedrake/petsc/config/BuildSystem/config/package.py", line 743, in checkDownload
    return self.getInstallDir()
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/firedrake/petsc/config/BuildSystem/config/package.py", line 545, in getInstallDir
    installDir = self.Install()
                 ^^^^^^^^^^^^^^
  File "/home/firedrake/petsc/config/BuildSystem/config/package.py", line 2081, in Install
    raise RuntimeError('Error running make on  '+self.PACKAGE)
================================================================================
Finishing configure run at Sat, 26 Apr 2025 13:10:03 +0100
================================================================================
</p>
</details> 

There have been recent changes on the Firedrake side, see https://github.com/firedrakeproject/firedrake/pull/4122.

--- a/docker/Dockerfile.firedrake-parmmg
+++ b/docker/Dockerfile.firedrake-parmmg
@@ -57,8 +57,8 @@ ENV OMPI_ALLOW_RUN_AS_ROOT=1 OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1
 RUN git clone --depth 1 --branch $(python3 firedrake-configure --show-petsc-version) https://gitlab.com/petsc/petsc.git \
     && cd petsc \
     && python3 /home/firedrake/firedrake-configure --show-petsc-configure-options | \
-        sed "s/$/ --COPTFLAGS='-O3 -mtune=generic' --CXXOPTFLAGS='-O3 -mtune=generic' --FOPTFLAGS='-O3 -mtune=generic'/" | \
-        xargs -L1 ./configure --with-make-np=4 --download-eigen --download-parmetis --download-mmg --download-parmmg \
+        sed "s/-march=native -mtune=native/-mtune=generic/g" | \
+        xargs -L1 ./configure --with-make-np=4 --download-eigen --download-metis --download-parmetis --download-mmg --download-parmmg \
     && make \
     && make check \
     && rm -rf ./**/externalpackages \

--- a/docker/Dockerfile.firedrake-parmmg
+++ b/docker/Dockerfile.firedrake-parmmg
@@ -32,26 +32,23 @@ RUN usermod -d /home/firedrake -m ubuntu \
     && echo "firedrake ALL=(ALL) NOPASSWD: ALL" >>/etc/sudoers \
     && ldconfig
 
-WORKDIR /home/firedrake
-
 # ------------------------------
 # Install Firedrake dependencies
 # ------------------------------
 # Download firedrake-configure
+
+USER firedrake
+WORKDIR /home/firedrake
 RUN curl -O https://raw.githubusercontent.com/firedrakeproject/firedrake/master/scripts/firedrake-configure
 
+USER root
 # Install system dependencies
 RUN apt-get update \
     && apt-get -y install \
         $(python3 firedrake-configure --show-system-packages) \
     && rm -rf /var/lib/apt/lists/*
 
-# ------------------------------
-# Install PETSc with dependencies for mesh adaptation
-# ------------------------------
-# OpenMPI will complain if mpiexec is invoked as root unless these are set
-ENV OMPI_ALLOW_RUN_AS_ROOT=1 OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1
-
+USER firedrake
 # We set the compiler optimisation flags manually here to remove the default of
 # '-march=native' which is not suitable for Docker images.
 RUN git clone --depth 1 --branch $(python3 firedrake-configure --show-petsc-version) https://gitlab.com/petsc/petsc.git \
@@ -78,14 +75,23 @@ ENV CFLAGS="-mtune=generic" CPPFLAGS="-mtune=generic"
 ENV MPICC=$CC
 ENV PATH="/home/firedrake/.local/bin:$PATH"
 
+# Hotfix for petsc4py build, see https://gitlab.com/petsc/petsc/-/issues/1759
+RUN echo 'Cython<3.1' > constraints.txt
+ENV PIP_CONSTRAINT=constraints.txt
+
+RUN pip cache remove petsc4py
+RUN pip install "$PETSC_DIR"/src/binding/petsc4py
+RUN git clone https://github.com/firedrakeproject/firedrake.git
+RUN pip install -r ./firedrake/requirements-build.txt
+
+
 # ------------------------------
 # Install Firedrake and mesh adaptation packages
 # ------------------------------
-USER firedrake
+
 
 # Install Firedrake
-RUN pip install --verbose --no-binary h5py --src . -e \
-        git+https://github.com/firedrakeproject/firedrake.git#egg=firedrake[check]
+RUN pip install --no-build-isolation --no-binary h5py -e firedrake/[check]
 
 # Install Thetis
 RUN pip install --verbose --src . -e git+https://github.com/thetisproject/thetis.git#egg=thetis

